File: android-app/app/src/main/java/com/bandite/sonicwalkscape/data/models/TourManifest.kt
package com.bandite.sonicwalkscape.data.models

data class TourManifest(
    val tourId: String,
    val language: String,
    val version: Int,
    val audio: List<AudioFile>,
    val images: List<ImageFile>,
    val subtitles: List<SubtitleFile>,
    val offlineMap: OfflineMap? = null
) {
    data class AudioFile(
        val pointId: String,
        val order: Int,
        val fileUrl: String,
        val fileSizeBytes: Long
    )

    data class ImageFile(
        val pointId: String,
        val fileUrl: String,
        val fileSizeBytes: Long
    )

    data class SubtitleFile(
        val pointId: String,
        val language: String,
        val fileUrl: String,
        val fileSizeBytes: Long
    )

    data class OfflineMap(
        val tilesUrlTemplate: String,
        val bounds: Bounds,
        val recommendedZoomLevels: List<Int>
    ) {
        data class Bounds(
            val north: Double,
            val south: Double,
            val east: Double,
            val west: Double
        )
    }

    val totalSizeBytes: Long
        get() = audio.sumOf { it.fileSizeBytes } +
                images.sumOf { it.fileSizeBytes } +
                subtitles.sumOf { it.fileSizeBytes }
}

File: mobile-app/ios/SonicWalkscape/SonicWalkscape/Models/TourManifest.swift
//
//  TourManifest.swift
//  SonicWalkscape
//
//  Tour manifest model for audio, subtitles, and offline map data
//

import Foundation

struct TourManifest: Codable {
    let tourId: String
    let language: String
    let version: Int
    let audio: [AudioFile]
    let images: [ImageFile]
    let subtitles: [SubtitleFile]
    let offlineMap: OfflineMap?

    struct AudioFile: Codable {
        let pointId: String
        let order: Int
        let fileUrl: String
        let fileSizeBytes: Int
    }

    struct ImageFile: Codable {
        let pointId: String?
        let fileUrl: String
        let fileSizeBytes: Int
    }

    struct SubtitleFile: Codable {
        let pointId: String
        let language: String
        let fileUrl: String
        let fileSizeBytes: Int
    }

    struct OfflineMap: Codable {
        let tilesUrlTemplate: String
        let bounds: Bounds
        let recommendedZoomLevels: [Int]

        struct Bounds: Codable {
            let north: Double
            let south: Double
            let east: Double
            let west: Double
        }
    }
}
